src/lib.rs

```diff
@@ -418,6 +369,74 @@ impl Game {
         Ok(())
     }
 
+    fn kill_player_near(&mut self, position: Position) -> Result<(), JsValue> {
+        let local_player_color = match &self.local_player_color {
+            None => return Ok(()), // not controlling anything
+            Some(c) => *c,
+        };
+
+        let mut killed_player: Option<&mut Player> = None;
+        let mut closest_distance = self.kill_distance;
+
+        for player in self.players.iter_mut() {
+            // TODO(can't kill impostors)
+            if player.color == local_player_color || player.dead {
+                continue;
+            }
+
+            let distance = position.distance(player.position);
+            if distance < closest_distance {
+                killed_player = Some(player);
+                closest_distance = distance;
+            }
+        }
+
+        if let Some(player) = killed_player {
+            player.dead = true;
+            let player: &Player = player;
+            let body = DeadBody {
+                position: player.position,
+                color: player.color,
+            };
+            self.bodies.push(body);
+            self.send_msg(&Message::Killed(body))?;
+            // Move the killer on top of the new body.
+            if let Some(player) = self.local_player_mut() {
+                player.position = body.position;
+            }
+        }
+
+        Ok(())
+    }
+
+    fn activate_near(&mut self, position: Position) -> Result<(), JsValue> {
+        let mut closest_distance = self.kill_distance;
```

Comment:
Should there be a distinct `activation_distance`?

Reply:
+1, done